scipy/stats/_mannwhitneyu.py

```diff
@@ -0,0 +1,367 @@
+    .. [4] Rosie Shier, "Statistics: 2.3 The Mann-Whitney U Test", Mathematics
+           Learning Support Centre, 2004.
+
+    Examples
+    --------
+    We follow the example from [4]_: nine randomly sampled young adults were
+    diagnosed with type II diabetes at the ages below.
+
+    >>> males = [19, 22, 16, 29, 24]
+    >>> females = [20, 11, 17, 12]
+
+    We use the Mann-Whitney U test to assess whether there is a statistically
+    significant difference in the age of diagnosis of males and females.
+    The null hypothesis is that the probability of a male diagnosis age being
+    greater than a female diagnosis age is equal to the probability of a
+    female diagnosis age being greater than a male diagnosis age. We decide
+    that a confidence level of 95% is required to reject the null hypothesis
+    in favor of the alternative: that either the diagnosis age of males is
+    stochastically greater than the diagnosis age of females or the diagnosis
+    age of females is stochastically greater than the diagnosis age of males.
+    Since the number of samples is very small and there are no ties in the
+    data, we can compare the test statistic against the *exact* distribution
+    of the test statistic under the null hypothesis.
+
+    >>> from scipy.stats import mannwhitneyu2
+    >>> U1, p = mannwhitneyu2(males, females, method="exact")
+    >>> print(U1)
+    17.0
+
+    `mannwhitneyu2` always reports the statistic associated with the first
+    sample, which, in this case, is males. This agrees with :math:`U_M = 17`
+    reported in [4]_. The statistic associated with the second statistic
+    can be calculated:
+
+    >>> nx, ny = len(males), len(females)
+    >>> U2 = nx*ny - U1
+    >>> print(U2)
+    3.0
+
+    This agrees with :math:`U_F = 3` reported in [4]_. The two-sided
+    *p*-value can be calculated from either statistic, and the value produced
+    by `mannwhitneyu2` agrees with :math:`p = 0.11` reported in [4]_.
+
+    >>> print(p)
+    0.1111111111111111
+
+    The exact distribution of the test statistic is asymptotically normal, so
+    the example continues by comparing the exact *p*-value against the
+    *p*-value produced using the normal approximation.
+
+    >>> _, pnorm = mannwhitneyu2(males, females, method="asymptotic")
+    >>> print(pnorm)
+    0.11134688653314041
+
+    Here `mannwhitneyu2`'s reported *p*-value appears to conflict with the
+    value :math:`p = 0.09` given in [4]_. The reason is that [4]_
+    does not apply the continuity correction performed by `mannwhitneyu2`;
+    `mannwhitneyu2` reduces the distance between the test statistic and the
+    mean :math:`\mu = n_x n_y / 2` by 0.5 to correct for the fact that the
+    discrete statistic is being compared against a continuous distribution.
+    Here, the :math:`U` statistic used is less than the mean, so we reduce
+    the distance by adding 0.5 in the numerator.
+
+    >>> import numpy as np
```

Comment:
import not needed as I learned in the CvM PR :)

Reply:
Yes, I've received this comment before, too, like the one above. I continue to include `print` and `import numpy as np` until I am forced to change them because I think the rationale I've heard for removing them (e.g. "not necessary", "saves space", "looks cleaner") are outweighed by the reason for including it (user convenience when the assumptions for it being unnecessary are not met). What do you think?